internal/backend/projects/lint.go

```diff
@@ -0,0 +1,108 @@
+package projects
+
+import (
+	"fmt"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/projectsgrpcsvc"
+	"go.autokitteh.dev/autokitteh/internal/manifest"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type Checker func(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation
+
+var (
+	lintCheckers []Checker
+)
+
+const manifestFile = "autokitteh.yaml"
+
+func init() {
+	lintCheckers = []Checker{
+		checkNoTriggers,
+		checkEmptyVars,
+	}
+}
+
+func Validate(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation {
+	var vs []*sdktypes.CheckViolation
+	for _, checker := range lintCheckers {
+		vs = append(vs, checker(projectID, manifest, resources)...)
+	}
+
+	return vs
+}
+
+func checkNoTriggers(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation {
+	if len(m.Project.Triggers) == 0 {
+		return []*sdktypes.CheckViolation{
+			{
+				FileName: manifestFile,
+				Level:    sdktypes.ViolationError,
+				Message:  "no triggers",
+			},
+		}
+	}
+
+	return nil
+}
+
+func checkEmptyVars(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation {
+	var vs []*sdktypes.CheckViolation
+	for _, v := range m.Project.Vars {
+		if v.Value == "" {
```

Comment:
are you sure we don't allow empty variables ?
sometimes I use empty vars as booleans, just to indicate if it exists or not

Reply:
it's a warning. but i'd like a way to disable it.